lib/phoenix_live_view/test/live_view_test.ex

```diff
@@ -1071,47 +1093,61 @@ defmodule Phoenix.LiveViewTest do
   @doc """
   Asserts a redirect will happen within `timeout`.
 
-  It returns the flash messages from said redirect, if any.
-  Note the flash will contain string keys.
+  It returns a tuple containing the new path and the flash messages
+  from said redirect, if any. Note the flash will contain string keys.
 
   ## Examples
 
       render_click(view, :event_that_triggers_redirect)
-      flash = assert_redirect view, "/path"
+      {_path, flash} = assert_redirect view, "/path"
```

Comment:
My thought was that we would not return the path if we match on it, so we keep backwards compatibility. What do you think? :)